**Zahara Morris - Property Sales Transactions**

Link - https://data.wprdc.org/dataset/real-estate-sales/resource/57b84145-f9eb-493b-a78f-9bc803413d99

The code below analyzes property sales for the year 2025. Since the year has not yet been completed, the code only collects what it can from the data, covering the period from the beginning of the year to now (January-November). It analyzes two different ranks, which are the top 15 school districts with the most property sales and the top 15 school districts that recorded the fewest sales this year so far. This code provides a list for each rank and displays the number of sales in each school district. If needed, this code can also be adjusted to list a different amount than 15. 

In [7]:
import pandas as pd

FILE_PATH = 'PropSalesDataSet.csv'

NEIGHBORHOOD_COL = 'SCHOOLDESC' 
SALE_DATE_COL = 'SALEDATE'

try:
    
    df = pd.read_csv(FILE_PATH)

    df[SALE_DATE_COL] = pd.to_datetime(df[SALE_DATE_COL], errors='coerce')

    df.dropna(subset=[NEIGHBORHOOD_COL, SALE_DATE_COL], inplace=True)

    df_2025_sales = df[df[SALE_DATE_COL].dt.year == 2025].copy()

    if df_2025_sales.empty:
        print("\n--- No 2025 Sales Found ---")
        print("The filtered dataset does not contain any transactions processed for the year 2025.")
        exit()

    sales_counts = df_2025_sales[NEIGHBORHOOD_COL].value_counts()

    # Most Sold
    top_15_most_sold = sales_counts.head(15).rename('2025 Sales Count')

    # Least Sold
    top_15_least_sold = sales_counts.sort_values(ascending=True).head(15).rename('2025 Sales Count')

    print("Top 15 School Destricts by Most Property Sales (Jan-Nov 2025 Data)")
    print(top_15_most_sold.to_string())
    print("\n" + "="*70 + "\n")
    print("Top 15 School Destricts by Least Property Sales (Jan-Nov 2025 Data)")
    print(top_15_least_sold.to_string())
    
except FileNotFoundError:
    print(f"Error: The file '{FILE_PATH}' was not found. Please check the file name.")
except Exception as e:
    print(f"An unexpected error occurred: {e}")

Top 15 School Destricts by Most Property Sales (Jan-Nov 2025 Data)
SCHOOLDESC
Pittsburgh           5939
North Allegheny      1052
Woodland Hills        952
Penn Hills Twp        840
Shaler Area           738
Chartiers Valley      725
North Hills           710
McKeesport Area       638
Montour               624
Baldwin Whitehall     609
West Allegheny        589
Fox Chapel Area       570
Gateway               552
Plum Boro             531
Moon Area             507


Top 15 School Destricts by Least Property Sales (Jan-Nov 2025 Data)
SCHOOLDESC
Penn-Trafford          1
Fort Cherry           13
Duquesne City        127
Cornell              141
Riverview            163
Brentwood Boro       187
Clairton City        201
Avonworth            205
Northgate            206
Allegheny Valley     228
Carlynton            256
South Park           269
Sto-Rox              274
Deer Lakes           303
Elizabeth Forward    312


In conclusion, the analysis of the property sale data set (Jan-Nov) provided the information that the most property sales active School District was the Pittsburgh School District, which has 5939 property sales this year (Jan-Nov). The least showed Penn-Trafford School District, which has only 1 property sale this year (Jan-Nov). These show the high volume vs low volume markets, for example, low volume school districts like Penn-Trafford, which include low volume neighborhoods like Trafford, Penn Township, Harrison City, and Manor, struggle to attract buyers to their area, which shows weakness in the community and may not be the place someone would want to settle down. While high-volume school districts like Pittsburgh have high-volume neighborhoods like Squirrel Hill, Shadyside, and Regent Square. Regent Square also stands out here just because it ranks high with other data sets, which implies that this would be a nice neighborhood to settle down in.